pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/pendingack/impl/PendingAckHandleImpl.java

```diff
@@ -946,6 +946,7 @@ public void completeHandleFuture() {
 
     public void exceptionHandleFuture(Throwable t) {
         final boolean completedNow = this.pendingAckHandleCompletableFuture.completeExceptionally(t);
+        persistentSubscription.getTopic().getSubscriptions().remove(subName);
```

Comment:
The subscription removal only can be executed when the pending ack store initialize failed(there have another place were calling `exceptionHandleFuture(t)`), right? And it only can be executed when the ZK node does not exist, right?